// symbology/dist/index.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.createFile = exports.createStream = exports.OutputType = exports.OutputOption = exports.EncodingMode = exports.DataMatrix = exports.SymbologyType = void 0;
var SymbologyType_1 = __importDefault(require("./types/enums/SymbologyType"));
var DataMatrix_1 = __importDefault(require("./types/enums/DataMatrix"));
var EncodingMode_1 = __importDefault(require("./types/enums/EncodingMode"));
var OutputOption_1 = __importDefault(require("./types/enums/OutputOption"));
var OutputType_1 = __importDefault(require("./types/enums/OutputType"));
var main_1 = require("./main");
var SymbologyType_2 = require("./types/enums/SymbologyType");
Object.defineProperty(exports, "SymbologyType", { enumerable: true, get: function () { return __importDefault(SymbologyType_2).default; } });
var DataMatrix_2 = require("./types/enums/DataMatrix");
Object.defineProperty(exports, "DataMatrix", { enumerable: true, get: function () { return __importDefault(DataMatrix_2).default; } });
var EncodingMode_2 = require("./types/enums/EncodingMode");
Object.defineProperty(exports, "EncodingMode", { enumerable: true, get: function () { return __importDefault(EncodingMode_2).default; } });
var OutputOption_2 = require("./types/enums/OutputOption");
Object.defineProperty(exports, "OutputOption", { enumerable: true, get: function () { return __importDefault(OutputOption_2).default; } });
var OutputType_2 = require("./types/enums/OutputType");
Object.defineProperty(exports, "OutputType", { enumerable: true, get: function () { return __importDefault(OutputType_2).default; } });
var main_2 = require("./main");
Object.defineProperty(exports, "createStream", { enumerable: true, get: function () { return main_2.createStream; } });
Object.defineProperty(exports, "createFile", { enumerable: true, get: function () { return main_2.createFile; } });
exports.default = {
    DataMatrix: DataMatrix_1.default,
    EncodingMode: EncodingMode_1.default,
    OutputOption: OutputOption_1.default,
    OutputType: OutputType_1.default,
    SymbologyType: SymbologyType_1.default,
    createStream: main_1.createStream,
    createFile: main_1.createFile,
    /** Legacy support - remove in next major release */
    Encoding: EncodingMode_1.default,
    Output: OutputOption_1.default,
    Barcode: SymbologyType_1.default
};
