// symbology/dist/types/enums/OutputOption.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var OutputOption;
(function (OutputOption) {
    /** No options selected. (default) */
    OutputOption[OutputOption["BARCODE_NO_ASCII"] = 1] = "BARCODE_NO_ASCII";
    /** Boundary bars above and below the symbol and between rows if stacking. */
    OutputOption[OutputOption["BARCODE_BIND"] = 2] = "BARCODE_BIND";
    /** Add a box surrounding the symbol and whitespace. */
    OutputOption[OutputOption["BARCODE_BOX"] = 4] = "BARCODE_BOX";
    /** Add a reader initialisation symbol to the data before encoding. */
    OutputOption[OutputOption["READER_INIT"] = 16] = "READER_INIT";
    /** Use a smaller font for the human readable text. */
    OutputOption[OutputOption["SMALL_TEXT"] = 32] = "SMALL_TEXT";
    /** Embolden the human readable text. */
    OutputOption[OutputOption["BOLD_TEXT"] = 64] = "BOLD_TEXT";
    /** Select the CMYK colour space option for encapsulated PostScript files. */
    OutputOption[OutputOption["CMYK_COLOUR"] = 128] = "CMYK_COLOUR";
    /** Plot a matrix symbol using dots rather than squares. */
    OutputOption[OutputOption["BARCODE_DOTTY_MODE"] = 256] = "BARCODE_DOTTY_MODE";
    /** Use GS instead FNC1 as GS1 separator. */
    OutputOption[OutputOption["GS1_GS_SEPARATOR"] = 512] = "GS1_GS_SEPARATOR";
})(OutputOption || (OutputOption = {}));
exports.default = OutputOption;
