// symbology/dist/types/enums/SymbologyType.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var SymbologyType;
(function (SymbologyType) {
    /** Code 11 */
    SymbologyType[SymbologyType["CODE11"] = 1] = "CODE11";
    /** Standard Code 2 of 5 */
    SymbologyType[SymbologyType["C25MATRIX"] = 2] = "C25MATRIX";
    /** Interleaved 2 of 5 */
    SymbologyType[SymbologyType["C25INTER"] = 3] = "C25INTER";
    /** Channel Code */
    SymbologyType[SymbologyType["CHANNEL"] = 140] = "CHANNEL";
    /** Codabar */
    SymbologyType[SymbologyType["CODABAR"] = 18] = "CODABAR";
    /** Code 128 (automatic subset switching) */
    SymbologyType[SymbologyType["CODE128"] = 20] = "CODE128";
    /** Code 128 (subset B) */
    SymbologyType[SymbologyType["CODE128B"] = 60] = "CODE128B";
    /** Code 2 of 5 IATA */
    SymbologyType[SymbologyType["C25IATA"] = 4] = "C25IATA";
    /** Code 2 of 5 Data Logic */
    SymbologyType[SymbologyType["C25LOGIC"] = 6] = "C25LOGIC";
    /** Code 2 of 5 Industrial */
    SymbologyType[SymbologyType["C25IND"] = 7] = "C25IND";
    /** Code 3 of 9 (Code 39) */
    SymbologyType[SymbologyType["CODE39"] = 8] = "CODE39";
    /** Code 32 */
    SymbologyType[SymbologyType["CODE32"] = 129] = "CODE32";
    /** Code 93 */
    SymbologyType[SymbologyType["CODE93"] = 25] = "CODE93";
    /** Deutshe Post Leitcode */
    SymbologyType[SymbologyType["DPLEIT"] = 21] = "DPLEIT";
    /** Deutshe Post Identcode */
    SymbologyType[SymbologyType["DPIDENT"] = 22] = "DPIDENT";
    /** EAN */
    SymbologyType[SymbologyType["EANX"] = 13] = "EANX";
    /** EAN + Check Digit */
    SymbologyType[SymbologyType["EANX_CHK"] = 14] = "EANX_CHK";
    /** EAN-14 */
    SymbologyType[SymbologyType["EAN14"] = 72] = "EAN14";
    /** Extended Code 3 of 9 (Code 39+) */
    SymbologyType[SymbologyType["EXCODE39"] = 9] = "EXCODE39";
    /** GS1-128 (UCC/EAN-128) */
    SymbologyType[SymbologyType["EAN128"] = 16] = "EAN128";
    /** GS1 DataBar-14 */
    SymbologyType[SymbologyType["RSS14"] = 29] = "RSS14";
    /** GS1 DataBar Expanded */
    SymbologyType[SymbologyType["RSS_EXP"] = 31] = "RSS_EXP";
    /** GS1 DataBar Limited */
    SymbologyType[SymbologyType["RSS_LTD"] = 30] = "RSS_LTD";
    /** HIBC Code 128 */
    SymbologyType[SymbologyType["HIBC_128"] = 98] = "HIBC_128";
    /** HIBC Code 39 */
    SymbologyType[SymbologyType["HIBC_39"] = 99] = "HIBC_39";
    /** ISBN (EAN-13 with verification stage) */
    SymbologyType[SymbologyType["ISBNX"] = 69] = "ISBNX";
    /** ITF-14 */
    SymbologyType[SymbologyType["ITF14"] = 89] = "ITF14";
    /** Korea Post */
    SymbologyType[SymbologyType["KOREAPOST"] = 77] = "KOREAPOST";
    /** LOGMARS */
    SymbologyType[SymbologyType["LOGMARS"] = 50] = "LOGMARS";
    /** MSI Plessey */
    SymbologyType[SymbologyType["MSI_PLESSEY"] = 47] = "MSI_PLESSEY";
    /** NVE-18 */
    SymbologyType[SymbologyType["NVE18"] = 75] = "NVE18";
    /** Pharmacode One-Track */
    SymbologyType[SymbologyType["PHARMA"] = 51] = "PHARMA";
    /** Plessey Code */
    SymbologyType[SymbologyType["PLESSEY"] = 86] = "PLESSEY";
    /** PZN */
    SymbologyType[SymbologyType["PZN"] = 52] = "PZN";
    /** Telepen Alpha */
    SymbologyType[SymbologyType["TELEPEN"] = 32] = "TELEPEN";
    /** Telepen Numeric */
    SymbologyType[SymbologyType["TELEPEN_NUM"] = 87] = "TELEPEN_NUM";
    /** UPC A */
    SymbologyType[SymbologyType["UPCA"] = 34] = "UPCA";
    /** UPC A + Check Digit */
    SymbologyType[SymbologyType["UPCA_CHK"] = 35] = "UPCA_CHK";
    /** UPC E */
    SymbologyType[SymbologyType["UPCE"] = 37] = "UPCE";
    /** UPC E + Check Digit */
    SymbologyType[SymbologyType["UPCE_CHK"] = 38] = "UPCE_CHK";
    /** Ultracode */
    SymbologyType[SymbologyType["ULTRA"] = 144] = "ULTRA";
    /** UPNQR (Univerzalni Plačilni Nalog QR) */
    SymbologyType[SymbologyType["UPNQR"] = 143] = "UPNQR";
    /** Vehicle Identification Number (United States) */
    SymbologyType[SymbologyType["VIN"] = 73] = "VIN";
    /** Aztec Code */
    SymbologyType[SymbologyType["AZTEC"] = 92] = "AZTEC";
    /** Aztec Runes */
    SymbologyType[SymbologyType["AZRUNE"] = 128] = "AZRUNE";
    /** Code One */
    SymbologyType[SymbologyType["CODEONE"] = 141] = "CODEONE";
    /** Data Matrix ECC200 */
    SymbologyType[SymbologyType["DATAMATRIX"] = 71] = "DATAMATRIX";
    /** DotCode */
    SymbologyType[SymbologyType["DOTCODE"] = 115] = "DOTCODE";
    /** Grid Matrix */
    SymbologyType[SymbologyType["GRIDMATRIX"] = 142] = "GRIDMATRIX";
    /** Han Xin (Chinese Sensible) Code */
    SymbologyType[SymbologyType["HANXIN"] = 116] = "HANXIN";
    /** HIBC Aztec Code */
    SymbologyType[SymbologyType["HIBC_AZTEC"] = 112] = "HIBC_AZTEC";
    /** Micro QR Code */
    SymbologyType[SymbologyType["MICROQR"] = 97] = "MICROQR";
    /** HIBC Data Matrix ECC200 */
    SymbologyType[SymbologyType["HIBC_DM"] = 102] = "HIBC_DM";
    /** HIBC QR Code */
    SymbologyType[SymbologyType["HIBC_QR"] = 104] = "HIBC_QR";
    /** Maxicode */
    SymbologyType[SymbologyType["MAXICODE"] = 57] = "MAXICODE";
    /** QR Code */
    SymbologyType[SymbologyType["QRCODE"] = 58] = "QRCODE";
    /** Australia Post Redirection */
    SymbologyType[SymbologyType["AUSREDIRECT"] = 68] = "AUSREDIRECT";
    /** Australia Post Reply Paid */
    SymbologyType[SymbologyType["AUSREPLY"] = 66] = "AUSREPLY";
    /** Australia Post Routing */
    SymbologyType[SymbologyType["AUSROUTE"] = 67] = "AUSROUTE";
    /** Australia Post Standard Customer */
    SymbologyType[SymbologyType["AUSPOST"] = 63] = "AUSPOST";
    /** Dutch Post KIX Code */
    SymbologyType[SymbologyType["KIX"] = 90] = "KIX";
    /** Japanese Postal Code */
    SymbologyType[SymbologyType["JAPANPOST"] = 76] = "JAPANPOST";
    /** Pharmacode Two-Track */
    SymbologyType[SymbologyType["PHARMA_TWO"] = 53] = "PHARMA_TWO";
    /** PLANET */
    SymbologyType[SymbologyType["PLANET"] = 82] = "PLANET";
    /** PostNet */
    SymbologyType[SymbologyType["POSTNET"] = 40] = "POSTNET";
    /** Royal Mail 4-State (RM4SCC) */
    SymbologyType[SymbologyType["RM4SCC"] = 70] = "RM4SCC";
    /** Royal Mail 4-State Mailmark */
    SymbologyType[SymbologyType["MAILMARK"] = 121] = "MAILMARK";
    /** USPS OneCode */
    SymbologyType[SymbologyType["ONECODE"] = 85] = "ONECODE";
    /** Codablock-F */
    SymbologyType[SymbologyType["CODABLOCKF"] = 74] = "CODABLOCKF";
    /** Code 16K */
    SymbologyType[SymbologyType["CODE16K"] = 23] = "CODE16K";
    /** Code 49 */
    SymbologyType[SymbologyType["CODE49"] = 24] = "CODE49";
    /** GS1 DataBar-14 Stacked */
    SymbologyType[SymbologyType["RSS14STACK"] = 79] = "RSS14STACK";
    /** GS1 DataBar-14 Stacked Omnidirectional */
    SymbologyType[SymbologyType["RSS14STACK_OMNI"] = 80] = "RSS14STACK_OMNI";
    /** GS1 DataBar Expanded Stacked */
    SymbologyType[SymbologyType["RSS_EXPSTACK"] = 81] = "RSS_EXPSTACK";
    /** HIBC PDF417 */
    SymbologyType[SymbologyType["HIBC_PDF"] = 106] = "HIBC_PDF";
    /** HIBC MicroPDF417 */
    SymbologyType[SymbologyType["HIBC_MICPDF"] = 108] = "HIBC_MICPDF";
    /** PDF417 */
    SymbologyType[SymbologyType["PDF417"] = 55] = "PDF417";
    /** PDF417 Truncated */
    SymbologyType[SymbologyType["PDF417TRUNC"] = 56] = "PDF417TRUNC";
    /** MicroPDF417 */
    SymbologyType[SymbologyType["MICROPDF417"] = 84] = "MICROPDF417";
    /** Composite symbol with EAN linear component */
    SymbologyType[SymbologyType["EANX_CC"] = 130] = "EANX_CC";
    /** Composite symbol with GS1-128 linear component */
    SymbologyType[SymbologyType["EAN128_CC"] = 131] = "EAN128_CC";
    /** Composite symbol with GS1 DataBar Limited component */
    SymbologyType[SymbologyType["RSS_LTD_CC"] = 133] = "RSS_LTD_CC";
    /** Composite symbol with GS1 DataBar Extended component */
    SymbologyType[SymbologyType["RSS_EXP_CC"] = 134] = "RSS_EXP_CC";
    /** Composite symbol with GS1 DataBar-14 Linear component */
    SymbologyType[SymbologyType["RSS14_CC"] = 132] = "RSS14_CC";
    /** Composite symbol with GS1 DataBar-14 Stacked component */
    SymbologyType[SymbologyType["RSS14STACK_CC"] = 137] = "RSS14STACK_CC";
    /** Composite symbol with GS1 DataBar-14 Stacked Omnidirectional component */
    SymbologyType[SymbologyType["RSS14_OMNI_CC"] = 138] = "RSS14_OMNI_CC";
    /** Composite symbol with GS1 DataBar Expanded Stacked component */
    SymbologyType[SymbologyType["RSS_EXPSTACK_CC"] = 139] = "RSS_EXPSTACK_CC";
    /** Composite symbol with UPC A linear component */
    SymbologyType[SymbologyType["UPCA_CC"] = 135] = "UPCA_CC";
    /** Composite symbol with UPC E linear component */
    SymbologyType[SymbologyType["UPCE_CC"] = 136] = "UPCE_CC";
    /** DAFT Code */
    SymbologyType[SymbologyType["DAFT"] = 93] = "DAFT";
    /** FIM */
    SymbologyType[SymbologyType["FIM"] = 49] = "FIM";
    /** Flattermarken */
    SymbologyType[SymbologyType["FLAT"] = 28] = "FLAT";
})(SymbologyType || (SymbologyType = {}));
exports.default = SymbologyType;
